src/app/COMPONENTS/task-details/task-details.component.ts

```diff
@@ -0,0 +1,48 @@
+import { Component, OnInit } from '@angular/core'
+import { Task } from '../../models/task.model'
+import { MaterialModule } from '../../material.module'
+import { TasksService } from '../../service/tasks.service'
+import { ActivatedRoute } from '@angular/router'
+import { Observable } from 'rxjs'
+import { CommonModule, NgIf } from '@angular/common'
+import { FormBuilder, FormControl, Validators, ReactiveFormsModule, FormGroup } from '@angular/forms'
+
+@Component({
+  selector: 'app-task-details',
+  standalone: true,
+  imports: [
+    MaterialModule,
+    CommonModule,
+    NgIf,
+    ReactiveFormsModule
+  ],
+  templateUrl: './task-details.component.html',
+  styleUrl: './task-details.component.scss'
+})
+export class TaskDetailsComponent implements OnInit {
+  ngOnInit() {
+    if (this.activatedRoute.snapshot.paramMap) {
+      this.taskId = this.activatedRoute.snapshot.paramMap.get('id')
```

Comment:
1 note, don't put methods above constructor, variables should be defined above the constructor, methods below. Also Instead of using snapshot, you can use activatedRoute.params to find id param and then use switchMap operator to fetch taskDetails (I will let you think about how to implement this, if you are having difficulties you can message me I will explain), this way you wont have to rely of an if statement to see if theres an id param, because what if there isnt? then there would be no data fetched because at that current time there was no id param present